R/flatten_data.R

```diff
@@ -1,6 +1,7 @@
 #' Flatten an ecocomDP dataset
 #' 
-#' @param tables (list of tbl_df, tbl, data.frame) A named list of ecocomDP tables.
+#' @param tables (list of tbl_df, tbl, data.frame) A named list of ecocomDP tables or an ecocomDP formatted dataset.
+#' @param dataset (list) An ecocomDP formatted dataset
```

Comment:
The way I modified this, it should be backward compatible with all existing code -- if you send a list of tables to `flatten_data` it will flatten the tables. If you use the arguments, you can either send a list of tables using the `tables` argument or an ecocomDP formatted dataset using `dataset` argument. 

Generally, I think we should have options for users to pass an ecocomDP dataset object to the flatten and plotting fxns. This is more intuitive to users and code using the ecocomDP library is a lot cleaner -- see the examples.

Reply:
You're right @sokole, no backwards incompatibility here. Thanks for pointing this out.

Regarding function inputs, I see how it would be easier if users could just pass the dataset blob object, or a flattened table, to any function and get a result. However, this implicit approach requires extra work by the code to validate the inputs and return a meaningful message to the user when issues are present. Please create these unexported functions, add to utilities.R, then call from `validate_arguments()`. Additionally, please include unit tests against foreseeable issues.